### ISSUE TITLE
Cannot navigate to intended page after login

### ISSUE THREAD

**Issue by daisyab** [systers/macc#23]
When user is logged out and inputs a URL that requires login credentials, user is taken to the login page.  After logging in, however, the application does not navigate to the intended page.

**Comment by rosariorobinson** [comment]
Daisy, is this just at the front page when logging into MACC itself? Has this been resolved? @Nerdylicious @maria

**Comment by daisyab** [comment]
@rosariorobinson yes, just when logging into the MACC.  The page should navigate to the URL that was requested after the user has logged in.  What happens is that the home page is loaded, then the user will have to manually navigate to the target page.

**Comment by Nerdylicious** [comment]
@daisyab 

Daisy, what do you mean by intended page?

After you login, you should get redirected automatically to the page that says "Peace Corps Hub" where you can choose from the Malaria and PeaceTrack sections.

**Comment by daisyab** [comment]
By intended page, I mean the page the user wants to access when he inputs a URL.

When the user is logged out and he types a URL ( for example, http://localhost:8001/malaria/list_posts/ and clicks on New Post), the user will be redirected to the login page and he will have to login.  After logging in, the user will be redirected to the Peace Corps Hub page where he will have to navigate his way to the create posts page.  
He should instead be redirected to the page where you are able to create posts which is his/her intended page.

**Comment by Nerdylicious** [comment]
@daisyab , the user is required to login to be able to access any of the features. The user should not be able to access any features unless they are logged in. For example, if you user goes to localhost:8001/malaria/list_posts/ without logging in, they **should** get redirected to the login page (because they aren't allowed to access features unless they have authenticated). I hope this makes sense!

**Comment by Neetuj** [comment]
I think what Daisy is saying is "After logging in, however, the application
does not navigate to the intended page"

Usually the UX is designed such that if a page requires user to login to be
able to render anything on that page the convention is
1) point the user to the login page
2) after user logs in successfully then take user to the pages he/she was
trying to access (without having to navigate to that page)

and what is happening here ( from what i understand .. i did not test it my
self)
0) say user puts this url localhost:8001/malaria/list_posts/  without
logging in
1) application brings up the login page since that page needs
authentication
2) once the authentication requirement is satisfied ( user has logged in)
.. does not take user back the intended page (url
localhost:8001/malaria/list_posts/) in this case but take user to  Peace
Corps Hub page ( i guess the home url)  and now user is left to navigate
their way to localhost:8001/malaria/list_posts/  from Peace Corps Hub page
which is not the correct UX design .. since user already gave the specific
place on the application on which he/she wants to land.
this might also lead to breaking some automated tests in future

cheers,
Neetu

**Comment by Neetuj** [comment]
Thanks for submitting the bug @daisyab   @Nerdylicious  can you confirm the bug by 
1) logging out 
2) accessing a URL (which is not the home/hub page) 

Once you do that we can decide if it needs to be closed or worked on  and to what priority.

**Comment by Nerdylicious** [comment]
@daisyab @Neetuj 

Thanks Neetu for clarifying this, I understand what the issue is now.

Daisy is right, the login page will always bring us to the main homepage each time. This issue needs to be worked on. What priority should we set this to? It's don't think this would be considered a critical bug so I would probably set it to low/medium priority.

**Comment by Neetuj** [comment]
yes .. we can call it a enhancement (for UX purposes) or a low priority bug

**Comment by rosariorobinson** [comment]
we have general labels for priority and issue type. check those first.

**Comment by medhach** [comment]
@Nerdylicious Is there any work pending on this issue?

**Comment by Nerdylicious** [comment]
@medhach I don't think anyone has worked on this issue yet. You can work on it if you would like.

**Comment by muddlebee** [comment]
@medhach I would like to work on this :)

**Comment by medhach** [comment]
@anweshknayak Sure go ahead!
